tests/conftest.py

```diff
@@ -158,6 +158,28 @@ def multidict_getversion_callable(multidict_module: ModuleType) -> Callable:
     return multidict_module.getversion
 
 
+def write(cls: Any, proto: int) -> bytes:
+    d = cls([("a", 1), ("a", 2)])
+    return pickle.dumps(d, proto)
+
+
+@pytest.fixture
+def in_memory_pickle_classes(multidict_module: ModuleType) -> dict[str, bytes]:
```

Comment:
It's best to depend on `any_multidict_class` and get the  tag/module name from `multidict_implementation`. So there's no need to have internal loops. The protocols could be moved into a dedicated fixture with `params=`.

The return value would be just the pickled bytes.

Reply:
Rewritten. If we're not using a dictionary there was no need for a tag-implementation level string so I simplified this. let me know if there's something else you had in mind.